ml_tools/feature_generator.py

```diff
@@ -0,0 +1,161 @@
+import pandas as pd
+from abc import ABC, abstractmethod
+
+
+class FeatureGenerator(ABC):
+
+    # def __init__(self, data):
+    #     self._data = data.copy()
+
+    @abstractmethod
+    def generate_feature(self, data, column=None, **kwargs):
+        raise NotImplementedError("generate_feature must be implemented")
+
+
+# class GenericGenerator(FeatureGenerator):
+#     def generate_feature(self, data, column=None):
+#         pass
+
+
+class Hour(FeatureGenerator):
+
+    @classmethod
```

Comment:
I would make sure to set @abstractmethods or whatever the equivalent is for @property in the base class to make sure name and feature_type are defined